python/tskit/util.py

```diff
@@ -183,6 +183,31 @@ def unpack_arrays(packed, offset):
     return ret
 
 
+def subset_ragged_col(packed, offset, subset, col="metadata"):
```

Comment:
This is a nice, elegant way of doing what we need! But, should it be public? Or is it too specific to our use case? I honestly don't know. It's a pretty thin wrapper around pack/unpack, but maybe a useful enough one? The return type is a bit odd. Hm - what if it returned a tuple, instead, wouldn't that work? So then we could do e.g. 
```
            self.individuals.set_columns(
                i.flags[indiv_to_keep],
                *util.subset_ragged_col(
                    i.location, i.location_offset, indiv_to_keep, "location"
                ),
                *util.subset_ragged_col(i.metadata, i.metadata_offset, indiv_to_keep),
            )
```
It would mean we couldn't use keywords, which are nice, but the return value would be more pythonic?  If we don't make this public (so, call it `_subset_ragged_col`) then we wouldn't have to test it.

In any case, I think the argument `col` should be called `name`, and the method should be `subset_ragged_column` (ie, with 'column' spelled out).

Reply:
Oh, and I think we should not provide a default to `col` - there's not a natural default, so the user should specify.

Reply:
changed the function and argument names. the problem with returning a tuple is that `set_columns` only takes keyword arguments.

Reply:
If you put the arguments in the right order, you don't need to specify the keywords?